pkg/operator/baremetal_pod.go

```diff
@@ -77,33 +128,95 @@ func setMariadbPassword() corev1.EnvVar {
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: ironicHtpasswdKey,
+			},
+		},
+	}
+}
+
+func generateRandomPassword() (string, error) {
 	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
 		"abcdefghijklmnopqrstuvwxyz" +
 		"0123456789")
 	length := 16
 	buf := make([]rune, length)
+	numChars := big.NewInt(int64(len(chars)))
 	for i := range buf {
-		buf[i] = chars[rand.Intn(len(chars))]
+		c, err := rand.Int(rand.Reader, numChars)
+		if err != nil {
+			return "", err
+		}
+		buf[i] = chars[c.Uint64()]
 	}
-	return (string(buf))
+	return string(buf), nil
 }
 
 func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig) error {
-	glog.V(3).Info("Checking if the Maridb password secret already exists")
+	glog.V(3).Info("Checking if the MariaDB password secret already exists")
 	_, err := client.Secrets(config.TargetNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
+		password, err := generateRandomPassword()
+		if err != nil {
+			return err
+		}
 		// Secret does not already exist. So, create one.
-		_, err := client.Secrets(config.TargetNamespace).Create(
+		_, err = client.Secrets(config.TargetNamespace).Create(
 			context.Background(),
 			&corev1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      baremetalSecretName,
 					Namespace: config.TargetNamespace,
 				},
 				StringData: map[string]string{
-					baremetalSecretKey: generateRandomPassword(),
+					baremetalSecretKey: password,
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		return err
+	}
+	return err
+}
+
+func createIronicPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig, name string, username string, configSection string) error {
+	glog.V(3).Info(fmt.Sprintf("Checking if the %s password secret already exists", name))
+	_, err := client.Secrets(config.TargetNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// Secret does not already exist. So, create one.
+		password, err := generateRandomPassword()
+		if err != nil {
+			return err
+		}
+		hash, err := bcrypt.GenerateFromPassword([]byte(password), 5) // Use same cost as htpasswd default
+		if err != nil {
+			return err
+		}
+		// Change hash version from $2a$ to $2y$, as generated by htpasswd.
+		// These are equivalent for our purposes.
+		hash[2] = 'y'
+
+		_, err = client.Secrets(config.TargetNamespace).Create(
+			context.Background(),
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: config.TargetNamespace,
+				},
+				StringData: map[string]string{
+					ironicUsernameKey: username,
+					ironicPasswordKey: password,
+					ironicHtpasswdKey: fmt.Sprintf("%s:%s", username, hash),
+					ironicConfigKey: fmt.Sprintf(
+						"[%s]\nauth_type=http_basic\nusername=%s\npassword=%s\n",
```

Comment:
With all of the string literals used for the keys, I expected to see this template defined as a multi-line string const as well.

Reply:
That'd probably be easier to read, even after you account for the weird indentation.